content/en/blog/_posts/2022-01-24-POV-Kubernetes-cluster-infrastructure-upgrade-strategies.md

```diff
@@ -0,0 +1,94 @@
+#### Important Points to note:
+1)  It must be noted that for blue green strategy to work well, both the clusters are to be maintained at all times. This means more infra structure cost and some additional maintenance overhead.
+2)	Each time the cluster is rebuilt due to infrastructure upgrade, the applications will have to be redeployed. Hence right effort needs to be considered for validations and application stakeholders’ involvement as needed.  
+3)	Due importance to be given to automate the deployment and validation process as much as possible to reduce the dependencies and any version mismatch.
+4)	Since there is a switch to a new environment each time, some application downtime can be noted depending on the type of the application. Example for an application with many synchronous transactions and orchestration needs, it may be important to ensure that a synchronous transaction initiated from an active cluster is completed in the same cluster before the switch.
+
+ 
+### In place upgrade
+
+In place upgrade is a strategy that ensures an upgrade without the need of maintaining or creating a new cluster.  
+In Kubernetes based platforms the upgrade is done for one node at a time and thereby upgrading the entire cluster.
+The node that is to be upgraded is marked as non-schedulable so that no new pods get deployed on it. The node is drained so that existing pods running on it have a graceful closure. The Kubernetes creates the equivalent pods on the other active nodes to ensure that the deployed applications continue to be available.
+Once it is confirmed that no pods are running on the node to be upgraded, the upgrade process is initiated. Once the node is upgraded, its again marked as schedulable so that the applications or services start getting deployed on the upgraded node.
+Depending on the type of the node to be updated (Master/Worker) and the components running on them, due care needs to be taken to ensure that nothing is missed, and all inter dependencies are taken care. Example: The etcd database on all master nodes must be in sync. Various Kubernetes based platforms available such as Red Hat Openshift has mechanisms/scripts to ensure these things.
+In place upgrade can be considered as a bit complex upgrade procedure as each node/component have to be upgraded and verified for completion.
+ ![image](https://user-images.githubusercontent.com/63296841/150771693-7a6a6675-f170-47de-8830-700c8e01e016.png)
+
+#### Advantages:
+1)	Is Cost effective compared to strategies like Blue Green, as there are a smaller number of servers to be maintained.
+2)	The applications are not redeployed from scratch. New pods relevant to the same application image are deployed by Kubernetes on the active nodes. Hence the applications related stakeholder involvement and dependency management may go away. 
+3)	Near to Zero Downtime 
```

Comment:
Please use Markdown ordered lists here. See https://commonmark.org/help/tutorial/10-nestedLists.html for a guide.